pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -780,6 +782,15 @@ func (s *Service) reconcileELBTags(lb *infrav1.ClassicELB, desiredTags map[strin
 	return nil
 }
 
+func (s *Service) getHealthCheckELBProtocol() *infrav1.ClassicELBProtocol {
```

Comment:
```
func (s *Service) getHealthCheckELBProtocol() *infrav1.ClassicELBProtocol {
	controlPlaneELB := s.scope.ControlPlaneLoadBalancer()
	if controlPlaneELB != nil && controlPlaneELB.HealthCheckProtocol != nil {
		return controlPlaneELB.HealthCheckProtocol
	}
	return *infrav1.ClassicELBProtocolSSL
}
```
In this way we can reduce one new variable named elbProtocol and handle nil case if HealthCheckProtocol is nil in LBSpec. I understand we are handing nil case in defaults webhooks, but IMO it's better to not rely only on webhooks.

Reply:
I added this before, but as per [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3124#discussion_r797313475) comment, I removed it.

Reply:
Then you can write it as 
```
func (s *Service) getHealthCheckELBProtocol() *infrav1.ClassicELBProtocol {
	controlPlaneELB := s.scope.ControlPlaneLoadBalancer()
	if controlPlaneELB != nil && controlPlaneELB.HealthCheckProtocol != nil {
		return controlPlaneELB.HealthCheckProtocol
	}
	return nil
}
```
to reduce one new variable named `elbProtocol`

Reply:
Done

Reply:
Done